src/Log/QueryLogger.php

```diff
@@ -30,7 +31,12 @@ public function getLogs(): array
      */
     public function log($level, string|Stringable $message, array $context = []): void
     {
-        $this->_logs[] = (string)$context['query'];
+        if ($context['query'] instanceof LoggedQuery) {
+            /** @psalm-suppress InternalMethod **/
+            $this->_logs[] = $context['query']->jsonSerialize();
```

Comment:
Looking at Cake 3.x the casting `LoggedQuery` to string gave the format `"duration={$this->took} rows={$this->numRows} {$this->query}"` as seen [here](https://github.com/cakephp/cakephp/blob/3.x/src/Database/Log/LoggedQuery.php#L67). So if you want to retain old format that's what needs to be done here instead of using the array provided by `jsonSerialized()`.

You can use the `LoggedQuery::getContext()` method to get the details and construct a string with the above format.

Reply:
Logging the string message in that format is handled by the parent call to the cake QueryLogger::log(). getLogs() always returned an array of dictionaries in the format that jsonSerialize() returns.

Reply:
Does this seem right to you? @ADmad

Reply:
In your initial comment you said you want to restore the original workings of the query logger. The changes I suggested to achieve the format used in Cake 3 compatible version of the plugin (since Cake 4 casing `LoggedQuery` to string just return the SQL query). `getLogs()` has always returned an array of string, not an array or arrays.

Reply:
It seems to me that the query logger did return an array of LoggedQueries actually, please refer to commit 9aa30db8aa68b4da453f5fc9b062732c7d7fc93d , where the _log method appends the LoggedQuery object to _logs[].

Reply:
Hmm.. I checked again and indeed that seems to be the case. So I guess all we need to do is remove the string casting and update the docblock of `$_logs` property to indicate it's an array of `LoggedQuery`.

BTW where are you seeing/using the array returned by `getLogs()`?  `ExceptionRenderer` and `ApiQueryLogListener` sets the array as a view var. When the vars are serialized for JSON/XML response the instances would be cast to string and you will still end up with just the SQL query and before. So what does this change achieve?

Reply:
We are using a custom `ApiQueryLogListener` that also returns (for every connection) the array of `LoggedQuery` objects. When accessing the page, a `queryLog` key is appended to the output, containing the query that was executed, the number of rows affected, parameters, how long the query took and any errors. It seems to me that this is the result of json serializing using the `jsonSerialize()` method in `LoggedQuery`.

<img width="129" alt="query log" src="https://github.com/FriendsOfCake/crud/assets/164882046/d591e93f-c4a5-48eb-897d-e21af6f82714">

Reply:
Okay, implemented that in the latest comit